### ISSUE TITLE
[BUG] private_key property is not returned when creating an api_key

### ISSUE THREAD

**Issue by baelen-git** [CiscoDevNet/terraform-provider-intersight#90]
**description**
When I do a POST on the [API_KEY endpoint](https://intersight.com/apidocs/apirefs/api/v1/iam/ApiKeys/post/) with the body ```{ "Purpose" : "test" }``` it will create a new key and in the ```response``` I will see the ```private_key``` property is filled.
When  I create a key from Terraform it doesn't return the private_key in the state file

**how to reproduce**
code:
```
resource "intersight_iam_api_key" "api_key" {
  purpose           = "test API key"
}
```
Look in the statefile after this, the private_key property is empty.

**Expected behavior**
The private key to show up in the terraform state

**Provider Version :** v1.0.7

**Comment by code-lucidal58** [comment]
The private key is a property whose value can be received only when it was created. Subsequent fetch calls will not have the value. Hence, accessing this property as `intersight_iam_key.private` might give you value for the first time, but subsequent runs will give you an empty string. 

After the creation of the resource, the provider sends out a fetch call again and stores that value in the statefile. So you will not receive the secret key ever. Suppose, the code is modified to skip the fetch call, you will get the secret key, only once. But the subsequent call of any terraform command (apply, plan, destroy, refresh) will over-write the value. Will this work for you?

Please note that, in general, there are many property values across MOs that are filled by the backend after a delay of 1-2 seconds. Skipping the fetch call will result in many of the properties as an empty value in the state file, unless explicitly refreshed using `terraform refresh`.

**Comment by meghana-rajashekar** [comment]
#### Fixed with the latest provider release: v1.0.27
